packages/components/src/HeaderBar/HeaderBar.component.js

```diff
@@ -260,22 +217,77 @@ HeaderBar.Help = Help;
 HeaderBar.User = User;
 HeaderBar.Products = Products;
 
-HeaderBar.propTypes = {
-	logo: PropTypes.shape(Logo.propTypes),
-	brand: PropTypes.shape(Brand.propTypes),
-	env: PropTypes.shape(Environment.propTypes),
-	search: PropTypes.shape(Search.propTypes),
-	help: PropTypes.shape(Help.propTypes),
-	user: PropTypes.shape(User.propTypes),
-	products: PropTypes.shape(Products.propTypes),
-	renderers: PropTypes.shape({
-		Logo: PropTypes.func,
-		Brand: PropTypes.func,
-		Environment: PropTypes.func,
-		Search: PropTypes.func,
-		User: PropTypes.func,
-		Products: PropTypes.func,
-	}),
-};
+if (process.env.NODE_ENV !== 'production') {
+	Logo.propTypes = {
+		isFull: PropTypes.bool,
+		renderers: PropTypes.shape({
+			Action: PropTypes.func,
+		}),
+		t: PropTypes.func.isRequired,
+	};
+
+	Brand.propTypes = {
+		isSeparated: PropTypes.bool,
+		renderers: PropTypes.shape({
+			Action: PropTypes.func,
+		}),
+	};
+
+	Environment.propTypes = {
+		renderers: PropTypes.shape({
+			ActionDropdown: PropTypes.func,
+		}),
+	};
+
+	Search.propTypes = {
+		...Typeahead.propTypes,
+		renderers: PropTypes.shape({
+			Typeahead: PropTypes.func,
+		}),
+	};
+
+	Help.propTypes = {
+		renderers: PropTypes.shape({
+			ActionSplitDropdown: PropTypes.func,
+			Action: PropTypes.func,
+		}),
+		t: PropTypes.func.isRequired,
+	};
+
+	User.propTypes = {
+		renderers: PropTypes.shape({
+			ActionDropdown: PropTypes.func,
+			name: PropTypes.string.isRequired,
+			firstName: PropTypes.string,
+			lastName: PropTypes.string,
+		}),
+	};
+
+	Products.propTypes = {
+		renderers: PropTypes.shape({
+			ActionDropdown: PropTypes.func,
+		}),
+		t: PropTypes.func.isRequired,
+	};
+
+	HeaderBar.propTypes = {
+		logo: PropTypes.shape(Logo.propTypes),
+		brand: PropTypes.shape(Brand.propTypes),
+		env: PropTypes.shape(Environment.propTypes),
+		search: PropTypes.shape(Search.propTypes),
+		help: PropTypes.shape(Help.propTypes),
+		user: PropTypes.shape(User.propTypes),
+		products: PropTypes.shape(Products.propTypes),
+		renderers: PropTypes.shape({
+			Logo: PropTypes.func,
+			Brand: PropTypes.func,
+			Environment: PropTypes.func,
+			Search: PropTypes.func,
+			User: PropTypes.func,
+			Products: PropTypes.func,
+		}),
+		t: PropTypes.func,
+	};
+}
 
-export default HeaderBar;
+export default translate(I18N_DOMAIN_COMPONENTS, { i18n: i18n.init() })(HeaderBar);
```

Comment:
We have to init the i18n else we haven't no translations if any provider is given.
https://github.com/i18next/i18next/blob/master/src/i18next.js#L247

There is not issue with the init on the provider because i18next extends the options with the existing options: https://github.com/i18next/i18next/blob/master/src/i18next.js#L37